src/integration/java/io/pinecone/helpers/BuildUpsertRequest.java

```diff
@@ -0,0 +1,60 @@
+package io.pinecone.helpers;
+
+import com.google.common.primitives.Floats;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import io.pinecone.proto.SparseValues;
+import io.pinecone.proto.UpsertRequest;
+import io.pinecone.proto.Vector;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class BuildUpsertRequest {
+    private static final float[][] upsertData = {{1.0F, 2.0F, 3.0F}, {4.0F, 5.0F, 6.0F}, {7.0F, 8.0F, 9.0F}};
+
+    public static UpsertRequest buildRequiredUpsertRequest() {
```

Comment:
This class has two buildUpsertRequest: One with required parameters only and the second one is with optional parameters

Reply:
Yes I'll need it for future upsert test.